framework/src/utils/MooseMeshUtils.C

```diff
@@ -283,27 +283,27 @@ getExtraIDUniqueCombinationMap(const MeshBase & mesh,
       MooseMeshUtils::getExtraIDUniqueCombinationMap(mesh, block_ids, extra_ids);
   // parsing extra ids based on ref_parsed_ids
   std::set<std::pair<dof_id_type, dof_id_type>> unique_ids;
+  std::map<std::pair<dof_id_type, dof_id_type>, std::set<const Elem *>> map_unique_id_to_elem;
   for (const auto & elem : mesh.active_element_ptr_range())
   {
     if (block_restricted && block_ids.find(elem->subdomain_id()) == block_ids.end())
       continue;
     const dof_id_type id1 = base_parsed_ids[elem->id()];
     const dof_id_type id2 = elem->get_extra_integer(id_index);
-    if (!unique_ids.count(std::pair<dof_id_type, dof_id_type>(id1, id2)))
-      unique_ids.insert(std::pair<dof_id_type, dof_id_type>(id1, id2));
+    const std::pair<dof_id_type, dof_id_type> ids = std::make_pair(id1, id2);
+    if (!unique_ids.count(ids))
+      unique_ids.insert(ids);
+    map_unique_id_to_elem[ids].insert(elem);
   }
   mesh.comm().set_union(unique_ids);
   std::map<dof_id_type, dof_id_type> parsed_ids;
-  for (const auto & elem : mesh.active_element_ptr_range())
+
+  for (const auto & it : map_unique_id_to_elem)
```

Comment:
Use a structured binding to construct meaningful names instead of needing to use `it.first` and `it.second`, e.g.
```suggestion
  for (const auto & [ids, elements] : map_unique_id_to_elem)
```